angular/src/app/views/settings/settings.component.ts

```diff
@@ -133,54 +172,61 @@ export class SettingsComponent {
    * @returns string
    */
   private validateSettings(): string {
-    let issueStringBuffer = "";
+    let issues = [];
 
-    if (this.stockBuffer < 0 || isNaN(this.stockBuffer)) {
-      issueStringBuffer +=
+    if (this.stockBuffer && (isNaN(this.stockBuffer) || this.stockBuffer < 0)) {
+      issues.push(
         this.translation.translate("buffer") +
-        ": " +
-        this.translation.translate(
-          SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
-        ) +
-        SettingsComponent.MESSAGE_DELIM;
+          ": " +
+          this.translation.translate(
+            SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
+          )
+      );
     }
 
-    if (this.defaultOrderStatus < 0 || isNaN(this.defaultOrderStatus)) {
-      issueStringBuffer +=
-        this.translation.translate("order_status_id") +
-        ": " +
-        this.translation.translate(
-          SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
-        ) +
-        SettingsComponent.MESSAGE_DELIM;
-    }
+    // defaultOrderStatus is being picked from list and no longer needs client-side validation.
 
     if (
       this.defaultShippingProvider &&
-      (this.defaultShippingProvider < 0 || isNaN(this.defaultShippingProvider))
+      (isNaN(this.defaultShippingProvider) || this.defaultShippingProvider < 0)
     ) {
-      issueStringBuffer +=
+      issues.push(
         this.translation.translate("shipping_provider_id") +
-        ": " +
-        this.translation.translate(
-          SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
-        ) +
-        SettingsComponent.MESSAGE_DELIM;
+          ": " +
+          this.translation.translate(
+            SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED
+          )
+      );
```

Comment:
```
let translationBuffer = this.translation.translate("buffer");
let translationNegativeNotAllowed = this.translation.translate(SettingsComponent.TRANSLATION_KEY_NEGATIVE_NOT_ALLOWED);
issues.push(`${translationBuffer}: ${translationNegativeNotAllowed}`)
```
Personal preference, but I think the interpolated string with those translations extracted out looks a little cleaner. Otherwise not a big deal

Reply:
I think I'm getting rid of that validation altogether by tightening up the UX to avoid allowing  invalid things to slip through - might be a new PR tomorrow with an overhaul.